### PR TITLE
Cog 1199aa

### PR DESCRIPTION
**Pull request by sorych**
### Tests
- [x] I have included unit tests 
- [x] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by dmytro-palczewski-osi** [comment]
I don't like this refactoring.
In my opinion our IdmServiceImpl service is all anout user so if we need dedicated User service in our system we may just rename IdmServiceImpl  to UserService.

**Comment by sorych** [comment]
@dmytro-palczewski-osi I cannot agree with you here, because after User-related logic is moved out to its own service with encapsulation of it's complexity, the IdmService, that is higher-level service became more clean and clear. IdmService uses validation and authorization services to check if the action can be performed while UserService is used to perform the action on User. Also, as it can be seen from refactoring results, the cwsUserInfoService, nsUserService, Mapping - all this was moved to UserService and can be used in other places (example - UniversalUserTocken Creation)

File: src/main/java/gov/ca/cwds/idm/service/UserService.java [hunk 1]
```diff
@@ -11,15 +41,158 @@
...
+    return new UsersPage(users, userPage.getPaginationToken());
+  }
+
+  public User enrichWithCwsData(final User userDto) {
+    CwsUserInfo cwsUser = getCwsUserData(userDto);
+    enrichUserByCwsData(userDto, cwsUser);
+    return userDto;
+  }
+
+  public User createUser(User userDto) {
```
Comment (dmytro-palczewski-osi) [inline src/main/java/gov/ca/cwds/idm/service/UserService.java]: Why does object of type User have misleading name userDto?

Comment (sorych) [reply src/main/java/gov/ca/cwds/idm/service/UserService.java]: This was a suggestion of @AlexanderSerbin that I agreed with. We use a User object here as a dto to transfer data (even not all data needed for user to be created) that is why the parameter was renamed

**Comment by dmytro-palczewski-osi** [comment]
> IdmService uses validation and authorization services to check if the action can be performed

I think that  validation and authorization should be made in scope of UserService operations.
In such case it is not clear what may be left in IdmService.

**Pull request merged**